tests/thread_priority_inversion/main.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2017 Technische Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Thread test application for priority inversion problem
+ *
+ * @author      Thomas Geithner <thomas.geithner@dai-labor.de>
+ *
+ * @}
+ */
+
+
+#include <stdio.h>
+#include <string.h>
+#include <stdint.h>
+
+#include "thread.h"
+#include "mutex.h"
+#include "xtimer.h"
+
+mutex_t res_mtx;
+
+char stack_high[THREAD_STACKSIZE_MAIN];
+char stack_mid[THREAD_STACKSIZE_MAIN];
+char stack_low[THREAD_STACKSIZE_MAIN];
+
+void *t_low_handler(void *arg)
+{
+    (void) arg;
+
+    /* starting working loop immediately */
+    while(1){
+        puts("t_low: allocating resource...");
+        mutex_lock(&res_mtx);
+        puts("t_low: got resource.");
+        xtimer_sleep(1);
+
+        puts("t_low: freeing resource...");
+        mutex_unlock(&res_mtx);
+        puts("t_low: freed resource.");
+        xtimer_sleep(1);
+    }
+    return NULL;
+}
+
+void *t_mid_handler(void *arg)
+{
+    (void) arg;
+
+    /* starting working loop after 3s */
+    xtimer_sleep(3);
+
+    puts("t_mid: doing some stupid stuff...");
+    while(1){
+        thread_yield_higher();
+    }
+}
+
+void *t_high_handler(void *arg)
+{
+    (void) arg;
+
+    /* starting working loop after 500 ms */
+    xtimer_usleep(500);
```

Comment:
this actually only 500 microseconds, if you want milliseconds change to: `500U * US_PER_MS`